feat: allow new insight type choice from sidebar

pauldambra opened the pull request:
## Problem

If you want to create a particular kind of new insight it can take over two clicks. Nobody has time for that

## Changes

Now it takes two clicks!

![2023-03-15 12 25 07](https://user-images.githubusercontent.com/984817/225309312-c036d981-5c61-47a2-ae27-985e8e339460.gif)

## How did you test this code?

👀 locally

pauldambra committed:
feat: allow new insight type choice from sidebar

pauldambra committed:
update e2e test

pauldambra committed:
add more e2e test fixes

Twixes commented on frontend/src/layout/navigation/SideBar/SideBar.tsx:
```diff
@@ -152,11 +157,25 @@ function Pages(): JSX.Element {
+                                onClickInside: hideSideBarMobile,
+                                overlay: (
+                                    <div className="SideBar__side-actions" data-attr="sidebar-new-insights">
+                                        <h5>New Insight</h5>
```
"New Insight" sounds really dry, how about something describing the _action_ being undertaken
```suggestion
                                        <h5>Creating a new insight</h5>
```

Twixes commented on frontend/src/layout/navigation/SideBar/SideBar.tsx:
```diff
@@ -152,11 +157,25 @@ function Pages(): JSX.Element {
                         identifier={Scene.SavedInsights}
                         to={urls.savedInsights()}
                         sideAction={{
-                            icon: <IconPlus />,
```
I think it's better to keep the plus button even if the _standard_ pattern is to use the arrow for dropdowns. The plus is just such a clear affordance here

Twixes commented on frontend/src/scenes/saved-insights/SavedInsights.tsx:
```diff
@@ -299,31 +280,9 @@ export function NewInsightButton({ dataAttr }: NewInsightButtonProps): JSX.Eleme
                     placement: 'bottom-end',
                     className: 'new-insight-overlay',
```
Wow, this dropdown is really  w i d e! I haven't notice before, because, well, I've always used the plus in the sidebar. But to ship this, let's constraint the width of `.new-insight-overlay` to something like 21rem (this number seems to result in all options having roughly equal-size two lines of description)

Twixes commented:
Ah, GitHub ate my review summary… Basically:
This doesn't seem to reduce the number of clicks? It used to be two and it's still two. In fact, it's now one click _more_ if you want a Trends insight.
Though we've talked about this idea before, and I think we came to the conclusion that there is one potential benefit: this can help users choose the most relevant insight type more conciously. So let's ship this. Just before we do this, I have a few comments.

pauldambra committed:
Update frontend/src/layout/navigation/SideBar/SideBar.tsx

Co-authored-by: Michael Matloka <dev@twixes.com>

pauldambra committed:
a plus and a width

pauldambra commented:
> This doesn't seem to reduce the number of clicks? It used to be two and it's still two. In fact, it's now one click more if you want a Trends insight.

I thought "Oh yeah, you can click the plus" 

I never do, which made me wonder if other people do. 

Last 7 days only 7% of clicks on the insight and plus area were on the plus

* 31% of people who click on the button area, click to save an insight
* 52% of people who click on the plus, then click to save an insight

Hmmm, numbers but maybe not insight :rofl

pauldambra commented:
<img width="566" alt="Screenshot 2023-03-15 at 19 25 54" src="https://user-images.githubusercontent.com/984817/225421054-4169d3e9-4758-4d2d-807d-2a3d33a2e9cf.png">

Twixes commented on frontend/src/layout/navigation/SideBar/SideBar.scss:
```diff
@@ -102,6 +102,10 @@
     max-height: calc(100vh - 20rem);
     max-width: calc(100vw - 14rem);
 
+    &[data-attr='sidebar-new-insights'] {
+        max-width: 21rem;
+    }
```
This style would also be useful on the "New insight" popover on the insights list page. Can we put it on `.new-insight-overlay` instead of this `SideBar` override?

pauldambra replied:
Hmmm, interesting, that already has an override that's being ignored

<img width="883" alt="Screenshot 2023-03-16 at 11 30 01" src="https://user-images.githubusercontent.com/984817/225603836-270620e9-60a5-4937-acc6-924935d17317.png">

*rolls up sleeves

pauldambra replied:
How about passing optionally passing max width to popover....

<img width="349" alt="Screenshot 2023-03-16 at 11 40 08" src="https://user-images.githubusercontent.com/984817/225606328-c230e80b-0084-4952-b190-cf3eb8cbc69a.png">
<img width="438" alt="Screenshot 2023-03-16 at 11 39 16" src="https://user-images.githubusercontent.com/984817/225606333-dad2cc29-4b09-4b23-986c-3b2b8a08c49b.png">
<img width="550" alt="Screenshot 2023-03-16 at 11 39 09" src="https://user-images.githubusercontent.com/984817/225606336-d812dcdb-284f-4473-8534-886ac19cb8a2.png">

pauldambra committed:
allow passing in max width to popover

pauldambra committed:
Merge branch 'master' into feat/new-insights-in-sidebar

Twixes commented on frontend/src/lib/lemon-ui/Popover/Popover.tsx:
```diff
@@ -120,7 +123,7 @@ export const Popover = React.forwardRef<HTMLDivElement, PopoverProps>(function P
                 apply({ availableWidth, availableHeight, rects, elements: { floating } }) {
                     Object.assign(floating.style, {
                         maxHeight: `${availableHeight}px`,
-                        maxWidth: `${availableWidth}px`,
```
So, the reason this is potentially dangerous is now we're not taking `availableWidth` into account – and we _always_ should be. But if we just `width` instead of `max-width` in `.new-insight-overlay`, this'd work well – that width would be used most of the time, but `max-width` would automatically take precedence when needed.

pauldambra replied:
🤔

pauldambra committed:
Revert "allow passing in max width to popover"

This reverts commit 415a751f71127ca5076126e209f92cd75a5b0024.

pauldambra committed:
third-ish times the charm

Twixes committed:
Merge branch 'master' into feat/new-insights-in-sidebar

Twixes committed:
Use a more concise tooltip

The tooltip basically functions as a on-hover label here, so using the same text as on the saved insights page, where it says "New insight".

Twixes reviewed (approved):
🚢

github-actions[bot] committed:
Update UI snapshots for `webkit` (2)

github-actions[bot] committed:
Update UI snapshots for `webkit` (2)

pauldambra committed:
Merge branch 'master' into feat/new-insights-in-sidebar

pauldambra commented:
Removing it in https://github.com/PostHog/posthog/pull/14876
